### PR TITLE
katago 1.6.1 -> 1.8.0

### PR DESCRIPTION
**Pull request by OmnipotentEntity**
###### Motivation for this change

Version update.  Support for online contribution.  [Added documentation to wiki.](https://nixos.wiki/wiki/Katago)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OmnipotentEntity** [comment]
Not ready to merge.  Trying to figure out how to properly get a git revision from a tag.  What I have doesn't work currently, and what I had before ofborg doesn't like.

**Comment by OmnipotentEntity** [comment]
I manually used the commit hash instead.  I've already spent several hours trying to save myself several seconds a month.

File: pkgs/games/katago/default.nix [hunk 1]
```diff
@@ -6,17 +6,20 @@
...
 , fetchFromGitHub
-, fetchpatch
+, fetchgit
+, writeShellScriptBin
 , cudnn ? null
 , cudatoolkit ? null
 , mesa ? null
 , opencl-headers ? null
 , ocl-icd ? null
+, openssl ? null
```
Comment (SuperSandro2000) [inline pkgs/games/katago/default.nix:16]:
```suggestion
, openssl
```

Despite being all over the place here this is not required if it is put behind lib.optional(s).

File: pkgs/games/katago/default.nix [hunk 2]
```diff
@@ -37,22 +40,28 @@ assert !enableCuda -> (
 assert enableTcmalloc -> (
   gperftools != null);
 
+assert enableContrib -> (
```
Comment (SuperSandro2000) [inline pkgs/games/katago/default.nix:43]: This is also not required.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
> I manually used the commit hash instead. I've already spent several hours trying to save myself several seconds a month.

This is a common way to do this.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109339` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>katago</li>
<li>katagoCPU</li>
<li>katagoWithCuda</li>
</ul>
</details>

File: pkgs/games/katago/default.nix (hunk 2 above)
Comment (OmnipotentEntity) [reply pkgs/games/katago/default.nix]: Thanks, I've adjusted this and the documentation.

File: pkgs/games/katago/default.nix (hunk 1 above)
Comment (OmnipotentEntity) [reply pkgs/games/katago/default.nix]: These too.

**Comment by OmnipotentEntity** [comment]
Gross.  fixing.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109339` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>katago</li>
<li>katagoCPU</li>
<li>katagoWithCuda</li>
</ul>
</details>

**Pull request merged**